### PR TITLE
addpkg: sharutils

### PR DESCRIPTION
**Pull request by qyl27**
Will still get a permission denied issue when packaging, but it does not affect its outputted result.  
That issue can be reproduced in x86 packaging, reported in https://gitlab.archlinux.org/archlinux/packaging/packages/sharutils/-/issues/2.  
Outdated config.guess issue was reported in https://savannah.gnu.org/support/index.php?111077.

### CODE REVIEW

File: sharutils/riscv64.patch [hunk 1]
```diff
@@ -0,0 +1,15 @@
+--- PKGBUILD
++++ PKGBUILD
+@@ -25,10 +25,12 @@ build() {
+ 	cd "${srcdir}/${pkgname}-${pkgver}"
+ 	CFLAGS+=' -fcommon' # https://wiki.gentoo.org/wiki/Gcc_10_porting_notes/fno_common
+ 	CFLAGS+=' -Wno-format-security' # fix build with gettext function
++	autoreconf -fi
```
Comment (CoelacanthusHex) [inline sharutils/riscv64.patch:7]: Should be in `prepare()` function.

File: sharutils/riscv64.patch [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+ 	./configure \
+ 		--prefix=/usr \
+ 		--mandir=/usr/share/man \
+ 		--infodir=/usr/share/info \
++		--disable-dependency-tracking
```
Comment (CoelacanthusHex) [inline sharutils/riscv64.patch:12]: Need a reason why we need it and why it's RISC-V specific.

**Review by CoelacanthusHex — COMMENTED** [review]
Lack the upstream report link on outdated config.guess.

**Comment by qyl27** [comment]
There is an error during building after `autoreconf`, it is an invalid argument of touch, after analyzing the output of automake, it is caused by the script selecting `touch` as the alternative to `autogen`.  
In archlinux upstream, they used the `config.guess` and `config.sub` which are in the source pack and outdated, that did not happenned, but after `autoreconf`, it appears due to enabling automatic dependency tracking by default.  
This issue was reported to upstream, see: https://savannah.gnu.org/support/index.php?111078 .  
(Thanks to @CoelacanthusHex , she is a good teacher.)

**Comment by qyl27** [comment]
The permission denied issue was fixed in makepkg.  
https://gitlab.archlinux.org/archlinux/packaging/packages/sharutils/-/issues/2 .

**Comment by CoelacanthusHex** [comment]
Please squash commits and update commit message to include what and why you did them, and add package version in the commit and PR title.

**Pull request closed without merging**